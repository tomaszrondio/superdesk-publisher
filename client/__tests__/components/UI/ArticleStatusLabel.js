import React from "react";
import ArticleStatusLabel from "../../../components/UI/ArticleStatusLabel";
import { render } from "@testing-library/react";

describe("UI/ArticleStatusLabel", () => {
  const publishedArticle = {
    status: "published",
    tenant: { name: "testtenant" },
    route: { name: "testroute" }
  };

  const unpublishedArticle = {
    status: "unpublished",
    tenant: { name: "testtenant" },
    route: { name: "testroute" }
  };

  it("renders published article labels correctly", () => {
    const { container, getByText } = render(
      <ArticleStatusLabel
        article={publishedArticle}
        url="https://sourcefabric.org"
      />
    );

    expect(getByText("testtenant")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders unpublished article labels correctly", () => {
    const { container, getByText } = render(
      <ArticleStatusLabel
        article={unpublishedArticle}
        url="https://sourcefabric.org"
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
