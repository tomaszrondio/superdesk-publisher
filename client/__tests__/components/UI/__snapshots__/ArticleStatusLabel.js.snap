// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/ArticleStatusLabel renders published article labels correctly 1`] = `
<a
  class="label label--success cursorPointer"
  href="https://sourcefabric.org"
  target="_blank"
>
  testtenant
  <span>
     / 
    testroute
  </span>
</a>
`;

exports[`UI/ArticleStatusLabel renders unpublished article labels correctly 1`] = `
<span
  class="label label--alert"
>
  testtenant
  <span>
     / 
    testroute
  </span>
</span>
`;
